File: 1-two-sum/two-sum.dart
// import 'dart:collections';
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    final Map<int, dynamic> hMap = Map();

    for (int i = 0; i < nums.length; i++){
        int curr = nums[i];
        int partner = target - curr;

        if (hMap.containsKey(partner)){
            return [hMap[partner], i];
        } else {
            hMap[curr] = i;
        }
    }
    return [];
  }
}

// hMap : #partnerleri tutacak
//     2: 0
//     11: 1
//     15: 2



// [2,11,15, 7], target = 9
//           ^
//     return [hMap[9-7],i]

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # holds each value with its index: {2:0, 7:1, 11:2, 15:3}
        pairMap = dict()

        for i, n in enumerate(nums):
            pairOfN = target - n

            if pairOfN in pairMap:
                return[pairMap[pairOfN], i]
            else:
                pairMap[n] = i

    
        